File: Manage_supervision/frontend/src/api/order.ts
import axios from '../utils/axios'

// 订单数据接口
export interface OrderDTO {
  id: number
  studentId: number
  teacherId: number
  courseId: number
  orderNumber: string
  price: number
  hours: number
  totalAmount: number
  status: string // PENDING, REFUND_PENDING, REFUND_REJECTED, CANCELED
  message: string
  refundReason: string
  rejectReason: string
  createTime: string
  updateTime: string
  
  // 关联信息
  studentName: string
  teacherName: string
  courseTitle: string
  courseSubject: string
  statusText?: string
}

// 订单状态中文映射
export const orderStatusMap = {
  'PENDING': '已支付',
  'REFUND_PENDING': '退款申请中',
  'REFUND_REJECTED': '退款已拒绝',
  'CANCELED': '已取消'
}

/**
 * 学生创建订单
 */
export const createOrder = async (
  courseId: number, 
  hours: number, 
  price: number, 
  message?: string
) => {
  try {
    console.log('创建订单参数:', { courseId, hours, price, message });
    
    // 确保courseId是有效的
    if (!courseId || isNaN(Number(courseId))) {
      throw new Error('课程ID无效');
    }
    
    // 使用JSON格式发送数据到专用的JSON端点
    const response = await axios.post('/api/student/orders/json', {
      courseId: courseId,
      hours: hours,
      price: price,
      message: message || ''
    });
    
    return response.data;
  } catch (error) {
    console.error('创建订单失败:', error);
    throw error;
  }
}

/**
 * 学生获取订单列表
 */
export const getStudentOrders = async () => {
  try {
    const response = await axios.get('/api/student/orders')
    return response.data
  } catch (error) {
    console.error('获取学生订单列表失败:', error)
    throw error
  }
}

/**
 * 学生获取订单详情
 */
export const getStudentOrderDetail = async (orderId: number) => {
  try {
    const response = await axios.get(`/api/student/orders/${orderId}`)
    return response.data
  } catch (error) {
    console.error('获取学生订单详情失败:', error)
    throw error
  }
}

/**
 * 学生申请退款
 */
export const cancelOrder = async (orderId: number, refundReason: string) => {
  try {
    // 使用/refund接口发送退款理由
    const response = await axios.post(`/api/student/orders/${orderId}/refund`, {
      refundReason: refundReason
    })
    return response.data
  } catch (error) {
    console.error('申请退款失败:', error)
    throw error
  }
}

/**
 * 教师获取订单列表
 */
export const getTeacherOrders = async () => {
  try {
    const response = await axios.get('/api/supervisor/orders')
    return response.data
  } catch (error) {
    console.error('获取教师订单列表失败:', error)
    throw error
  }
}

/**
 * 教师获取订单详情
 */
export const getTeacherOrderDetail = async (orderId: number) => {
  try {
    const response = await axios.get(`/api/supervisor/orders/${orderId}`)
    return response.data
  } catch (error) {
    console.error('获取教师订单详情失败:', error)
    throw error
  }
}

/**
 * 教师接受订单
 */
export const acceptOrder = async (orderId: number) => {
  try {
    const response = await axios.post(`/api/supervisor/orders/${orderId}/accept`)
    return response.data
  } catch (error) {
    console.error('接受订单失败:', error)
    throw error
  }
}

/**
 * 教师拒绝订单
 */
export const rejectOrder = async (orderId: number, rejectReason: string) => {
  try {
    const formData = new FormData()
    formData.append('rejectReason', rejectReason)
    
    const response = await axios.post(`/api/supervisor/orders/${orderId}/reject`, formData)
    return response.data
  } catch (error) {
    console.error('拒绝订单失败:', error)
    throw error
  }
}

/**
 * 教师完成订单
 */
export const completeOrder = async (orderId: number) => {
  try {
    const response = await axios.post(`/api/supervisor/orders/${orderId}/complete`)
    return response.data
  } catch (error) {
    console.error('完成订单失败:', error)
    throw error
  }
}

/**
 * 教师处理退款申请 - 同意
 */
export const approveRefund = async (orderId: number) => {
  try {
    const response = await axios.post(`/api/supervisor/orders/${orderId}/approve-refund`)
    return response.data
  } catch (error) {
    console.error('同意退款失败:', error)
    throw error
  }
}

/**
 * 教师处理退款申请 - 拒绝
 */
export const rejectRefund = async (orderId: number, rejectReason: string) => {
  try {
    // 使用JSON格式发送请求
    const response = await axios.post(`/api/supervisor/orders/${orderId}/reject-refund`, {
      rejectReason: rejectReason
    })
    return response.data
  } catch (error) {
    console.error('拒绝退款失败:', error)
    throw error
  }
} 

File: Manage_supervision/backend/src/main/java/com/example/auth/model/entity/Order.java
package com.example.auth.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单实体类
 */
@Data
@TableName("orders")
public class Order {
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField("student_id")
    private Long studentId;

    @TableField("teacher_id")
    private Long teacherId;

    @TableField("course_id")
    private Long courseId;

    @TableField("order_number")
    private String orderNumber;

    @TableField("price")
    private BigDecimal price;

    @TableField("hours")
    private Integer hours;

    @TableField("total_amount")
    private BigDecimal totalAmount;

    @TableField("status")
    private String status; // PENDING, ACCEPTED, REJECTED, CANCELED, COMPLETED, REFUND_PENDING, REFUND_REJECTED, APPEALING, APPEAL_APPROVED, APPEAL_REJECTED

    @TableField("message")
    private String message;

    @TableField("reject_reason")
    private String rejectReason;

    @TableField("refund_reason")
    private String refundReason;
    
    @TableField("appeal_reason")
    private String appealReason;
    
    @TableField("teacher_response")
    private String teacherResponse;

    @TableField("create_time")
    private LocalDateTime createTime;

    @TableField("update_time")
    private LocalDateTime updateTime;
    
    @TableField("appeal_time")
    private LocalDateTime appealTime;
    
    @TableField("admin_decision")
    private String adminDecision;

    // 非数据库字段，用于展示
    @TableField(exist = false)
    private String studentName;

    @TableField(exist = false)
    private String teacherName;

    @TableField(exist = false)
    private String courseTitle;

    @TableField(exist = false)
    private String courseSubject;
} 